src/transformers/models/chameleon/modeling_chameleon.py

```diff
@@ -1602,10 +2021,6 @@ def forward(
         logits = self.lm_head(hidden_states)
         logits = logits.float()
 
-        # Disallow image tokens which does not include special begin-image and end-image tokens
-        image_tokens = self.model.vocabulary_mapping.image_tokens
-        logits[:, :, image_tokens] = torch.finfo(logits.dtype).min
-
```

Comment:
here labels not update, maybe we should udpate labels here for calculate ce loss. In addition, in order to deal with different numbers of images in each sample in a batch, my suggestion is to reshape `input_ids` to fill `image_tokens` through the view method.

Below I provide a draft code I used

```python
if pixel_values is not None:
    batch_size, sequence_length = input_ids.shape
    input_ids = input_ids.view(batch_size * sequence_length)
    image_tokens = self.model.get_image_tokens(pixel_values)
    special_image_mask = input_ids == self.vocabulary_mapping.image_token_id
    image_tokens = image_tokens.to(input_ids.device, input_ids.dtype)
    input_ids = input_ids.masked_scatter(special_image_mask, image_tokens)
    input_ids = input_ids.view(batch_size, sequence_length)

if labels is not None:
    # update labels with new input_ids
    mask = labels != -100
    labels = torch.where(mask, input_ids, labels)
```